src/connectors/archive.js

```diff
@@ -39,7 +39,29 @@ Connector.getArtistTrack = () => {
 		track = trackParts[2];
 	}
 
+	if (hasAllTracksNumericPrefix('.jwrowV2 .ttl')) {
+		track = track.trim().replace(/^(\d+\w+)/, '');
+	}
+
 	return { artist, track };
 };
 
 Connector.albumSelector = '.thats-left > h1 [itemprop=name]';
+
+function hasAllTracksNumericPrefix(trackSelector) {
+	const tracks = document.querySelectorAll(trackSelector);
+	if (tracks.length === 0) {
+		return false;
+	}
+
+	let hasAllTracksNumericPrefix = true;
+	const re = /^\d+\w+/;
+	for (let i = 0; i < tracks.length; ++i) {
```

Comment:
`tracks` is NodeList, which is iterable, so please consider using `for..of` instead.

Reply:
Always forget this one